lib/dor/was_seed/thumbnail_generator_service.rb

```diff
@@ -7,32 +7,29 @@ class ThumbnailGeneratorService
       # because this date is earlier than any of the archived dates of the content,
       # this tells openwayback to provide the earliest capture date.
       DATE_TO_TRIGGER_EARLIEST_CAPTURE_DATE = '19900101120000'.freeze
-      def self.capture_thumbnail(druid, workspace, uri)
-        thumbnail_file = "#{DruidTools::Druid.new(druid, workspace).content_dir}/thumbnail.jp2"
-        wayback_uri    = "#{Settings.was_seed.wayback_uri}/#{DATE_TO_TRIGGER_EARLIEST_CAPTURE_DATE}/#{uri}"
-        temporary_file = "tmp/#{druid[6, 14]}"
-        jpeg_file = "#{temporary_file}.jpeg"
+      def self.capture_thumbnail(druid, workspace, seed_uri)
+        screenshot_jpeg = "tmp/#{druid[6, 14]}.jpeg"
         begin
-          indexed?(uri)
-          capture(wayback_uri, jpeg_file)
+          indexed?(seed_uri)
+          wayback_uri = "#{Settings.was_seed.wayback_uri}/#{DATE_TO_TRIGGER_EARLIEST_CAPTURE_DATE}/#{seed_uri}"
+          screenshot(wayback_uri, screenshot_jpeg)
+          resize_jpeg(screenshot_jpeg)
+          thumbnail_file = "#{DruidTools::Druid.new(druid, workspace).content_dir}/thumbnail.jp2"
+          Assembly::Image.new(screenshot_jpeg).create_jp2(output: thumbnail_file)
+          FileUtils.rm(screenshot_jpeg, force: true)
```

Comment:
resize, create_jp2 and removing the screenshot jpeg were all moved inside the block as all but `FileUtils.rm` would also result in the lack of a thumbnail, and I've turned off errors for the FileUtils.rm.